### PR TITLE
Translate radio/checkbox values when using bp_get_member_profile_data()

### PR DESCRIPTION
**Pull request by dgwatkins**
Took the chance to:

- fix the textdomain for groups.
- add a trailing slash to the member URL when required.

https://onthegosystems.myjetbrains.com/youtrack/issue/bpml-13

### CODE REVIEW

**Comment by dgwatkins** [comment]
@OnTheGoSystems/compatibility Please review

File: includes/class.groups.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
 class Groups implements \IWPML_Backend_Action, \IWPML_Frontend_Action {
 
 	const FIELDS     = [ 'name', 'description' ];
-	const TEXTDOMAIN = 'bpml';
+	const TEXTDOMAIN = 'Buddypress Multilingual';
```
Comment (strategio) [inline includes/class.groups.php:11]: @dgwatkins, we never released a version with the previous `bpml` text domain, right?

File: includes/class.xprofile.php [hunk 2]
```diff
@@ -292,6 +293,23 @@ public function t_group_name( $group_name ) {
                 "{$this->_group_string_prefix}{$group_id} name" ) : $group_name;
     }
 
+    public function t_data( $data, $r ) {
```
Comment (strategio) [inline includes/class.xprofile.php]: Could you please add a PHPDoc?

And also, the method name `t_data` and the `$r` variable are not very meaningful, could we improve it? Or is there a naming pattern that I am not aware of, in this project?

File: includes/class.xprofile.php [hunk 3]
```diff
@@ -292,6 +293,23 @@ public function t_group_name( $group_name ) {
...
+        $field_id = xprofile_get_field_id_from_name( $r['field'] );
+        if ( $field_id ) {
+            $field_type = bp_xprofile_get_field_type( $field_id );
```
Comment (strategio) [inline includes/class.xprofile.php:309]: @dgwatkins, [bp_xprofile_get_field_type](https://wbremedy.com/reference/functions/bp_xprofile_get_field_type/) can return `null`. You should check the return value in the `if` condition.

File: includes/class.xprofile.php [hunk 4]
```diff
@@ -292,6 +293,23 @@ public function t_group_name( $group_name ) {
...
+            if ( in_array( $field_type->field_obj->type, array( 'radio', 'checkbox', 'selectbox', 'multiselectbox' ) ) ) {
+                $data = stripslashes( apply_filters(
+                    'wpml_translate_single_string',
+                    $data,
+                    $this->_context,
```
Comment (strategio) [inline includes/class.xprofile.php:314]: If you decide to unify the text domain, then you should also "link" `WPML\BuddyPress\Groups::TEXTDOMAIN` and `$this->_context`.

Maybe:

```php
class BPML_XProfile implements \IWPML_Backend_Action, \IWPML_Frontend_Action {

    protected $_context = \WPML\BuddyPress\Groups::TEXTDOMAIN;

   //...

}
```

**Review by strategio — CHANGES_REQUESTED** [review]

File: includes/class.groups.php (hunk 1 above)
Comment (dgwatkins) [reply includes/class.groups.php:11]: Thats right

File: includes/class.xprofile.php (hunk 2 above)
Comment (dgwatkins) [reply includes/class.xprofile.php]: Added PHPDoc and renamed the `$r` argument too.

All the methods that translate in this class are named `t_xxxx`. I think its clear once you see the whole class.

File: includes/class.xprofile.php (hunk 3 above)
Comment (dgwatkins) [reply includes/class.xprofile.php:309]: Good catch, done

File: includes/class.xprofile.php (hunk 4 above)
Comment (dgwatkins) [reply includes/class.xprofile.php:314]: Makes sense, linked.

**Review by strategio — APPROVED** [review]
+1 @dgwatkins

**Review by kkarpieszuk — APPROVED** [review]

**Pull request merged**